### PR TITLE
feat: Generate selector for click events.

### PR DESCRIPTION
**Pull request by kinyoklion**
Prototype implementation.

### CODE REVIEW

File: packages/telemetry/browser-telemetry/src/collectors/dom/toSelector.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  let ptr = element;
+  while (isNode(ptr) && ptr.parentNode) {
+    const asString = elementToString(ptr as BasicElement);
+    // We do not need to include the 'html' component in the selector.
+    if (asString === 'html') {
+      break;
+    }
+    if (components.length > options.maxDepth) {
+      break;
+    }
```
Comment (tanderson-ld) [inline packages/telemetry/browser-telemetry/src/collectors/dom/toSelector.ts:132]: I think this condition could be moved to be in the while condition and would avoid an extra `elementToString` on the last iteration.

File: packages/telemetry/browser-telemetry/src/collectors/dom/toSelector.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  // Limiting the components should generate reasonable selectors in most cases.
+  const components: string[] = [];
+  let ptr = element;
+  while (isNode(ptr) && ptr.parentNode) {
+    const asString = elementToString(ptr as BasicElement);
+    // We do not need to include the 'html' component in the selector.
+    if (asString === 'html') {
+      break;
+    }
+    if (components.length > options.maxDepth) {
```
Comment (tanderson-ld) [inline packages/telemetry/browser-telemetry/src/collectors/dom/toSelector.ts:130]: If length is max depth, another push will happen.  Is that expected?

I guess it depends on if maxDepth includes the top element.  If that is the case, it is a little unexpected.  I would expect whatever option I provide to be the number of components in the returned string.

File: packages/telemetry/browser-telemetry/src/collectors/dom/toSelector.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+): string {
+  // For production we may want to consider if we additionally limit the maximum selector length.
+  // Limiting the components should generate reasonable selectors in most cases.
+  const components: string[] = [];
+  let ptr = element;
+  while (isNode(ptr) && ptr.parentNode) {
+    const asString = elementToString(ptr as BasicElement);
+    // We do not need to include the 'html' component in the selector.
+    if (asString === 'html') {
+      break;
```
Comment (tanderson-ld) [inline packages/telemetry/browser-telemetry/src/collectors/dom/toSelector.ts:128]: Should this be a continue?

**Review by tanderson-ld — CHANGES_REQUESTED** [review]

File: packages/telemetry/browser-telemetry/src/collectors/dom/toSelector.ts (hunk 3 above)
Comment (kinyoklion) [reply packages/telemetry/browser-telemetry/src/collectors/dom/toSelector.ts:128]: No. We want to stop if we get to HTML and not include it. Basically HTML is redundant information at the top.

**Review by tanderson-ld — APPROVED** [review]

**Pull request merged**